### PR TITLE
schedule: refine the operator priority

### LINKED ISSUE #5388: Support more accurate operator priority
## Enhancement Task
At present, we indeed have the priority for operators. But many kinds of operators share the high priority, which means they cannot replace each other. We need to tell these kinds of operators apart from the priority. e.g. The admin operator should have the highest priority.

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What problem does this PR solve?

Issue Number: Close #5388

### What is changed and how does it work?

This PR does the following things:
1. For admin operators, we use Urgent priority. These operators will also pass the store limit check.
2. For shuffle/random operators, which are used in the testing environment, we use Low priority.
3. For hot/replica/evict/grant operators, we use High priority.
4. The others use Medium priority.

```commit-message
```

### Check List

Tests

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

File: server/core/kind.go [hunk 1]
```diff
@@ -19,9 +19,10 @@ type PriorityLevel int
 
 // Built-in priority level
 const (
-	LowPriority PriorityLevel = iota
-	NormalPriority
-	HighPriority
+	Low PriorityLevel = iota
+	Medium
+	High
+	VeryHigh
```
[the remark this answers is not in the document]
Comment (rleungx) [reply server/core/kind.go:25]: Ok, I prefer Urgent

File: server/schedulers/grant_leader.go [hunk 2]
```diff
@@ -252,7 +252,7 @@ func (s *grantLeaderScheduler) Schedule(cluster schedule.Cluster, dryRun bool) (
 			continue
 		}
 		op.Counters = append(op.Counters, schedulerCounter.WithLabelValues(s.GetName(), "new-operator"))
-		op.SetPriorityLevel(core.HighPriority)
+		op.SetPriorityLevel(core.High)
```
Comment (bufferflies) [inline server/schedulers/grant_leader.go:255]: `grant leader` and `grant hot leader` are used in test, so the priority should be same with the shuffle.

File: server/schedulers/evict_leader.go [hunk 3]
```diff
@@ -348,7 +348,7 @@ func scheduleEvictLeaderOnce(name, typ string, cluster schedule.Cluster, conf ev
 			log.Debug("fail to create evict leader operator", errs.ZapError(err))
 			continue
 		}
-		op.SetPriorityLevel(core.HighPriority)
+		op.SetPriorityLevel(core.High)
```
Comment (bufferflies) [inline server/schedulers/evict_leader.go:351]: I think the priority of `evict leader` should be urgent, It should be higher than the hot region.

File: server/schedulers/grant_leader.go (hunk 2 above)
Comment (rleungx) [reply server/schedulers/grant_leader.go:255]: IMO, grant leader can also be used in the production environment.

File: server/schedulers/evict_leader.go (hunk 3 above)
Comment (rleungx) [reply server/schedulers/evict_leader.go:351]: make sense

File: server/schedule/operator_controller.go [hunk 4]
```diff
@@ -831,6 +831,10 @@ func (oc *OperatorController) ExceedStoreLimit(ops ...*operator.Operator) bool {
 
 // exceedStoreLimitLocked returns true if the store exceeds the cost limit after adding the operator. Otherwise, returns false.
 func (oc *OperatorController) exceedStoreLimitLocked(ops ...*operator.Operator) bool {
+	// The operator with Urgent priority, like admin operators, should ignore the store limit check.
+	if len(ops) != 0 && ops[0].GetPriorityLevel() == core.Urgent {
```
Comment (bufferflies) [inline server/schedule/operator_controller.go:835]: the condition `len(ops)>0` can be ignored because ops can not be nil in this step,.

**Review by bufferflies — APPROVED** [review]
LGTM

**Review by nolouch — APPROVED** [review]

**Pull request merged**